client/src/featureform/serving.py

```diff
@@ -424,9 +447,129 @@ def __init__(self, stream, dataframe=None):
+        provider and calculated and serving time.
+
+        **Examples**:
+
+        ``` py
+        import featureform as ff
+        client = ff.Client()
+        train, test = client
+            .training_set("fraud_training", "v1")
+            .train_test_split(
+                test_size=0.7,
+                train_size=0.3,
+                shuffle=True,
+                random_state=None,
+                batch_size=5
+            )
+
+        for features, label in train:
+            print(features)
+            print(label)
+            clf.partial_fit(features, label)
+
+        for features, label in test:
+            print(features)
+            print(label)
+            clf.score(features, label)
+
+
+        # TRAIN OUTPUT
+        # np.array([
+        #   [1, 1, 3],
+        #   [5, 1, 2],
+        #   [7, 6, 5],
+        #   [8, 3, 3],
+        #   [5, 2, 2],
+        # ])
+        # np.array([2, 4, 2, 3, 4])
+        # np.array([
+        #   [3, 1, 2],
+        #   [5, 4, 5],
+        # ])
+        # np.array([6, 7])
+
+        # TEST OUTPUT
+        # np.array([
+        #   [5, 1, 3],
+        #   [4, 3, 1],
+        #   [6, 6, 7],
+        # ])
+        # np.array([4, 6, 7])
+
+        ```
+
+        Args:
+            test_size (float): The ratio of test set size to train set size. Must be a value between 0 and 1. If excluded
+                it will be the complement to the train_size. One of test_size or train_size must be specified.
+            train_size (float): The ratio of train set size to train set size. Must be a value between 0 and 1. If excluded
+                it will be the complement to the test_size. One of test_size or train_size must be specified.
+            shuffle (bool): Whether to shuffle the dataset before splitting.
+            random_state (Union[int, None]): A random state to shuffle the dataset. If None, the dataset will be shuffled
+                randomly on every call. If >0, the value will be used a seed to create random shuffle that can be repeated
+                if subsequent calls use the same seed.
+            batch_size (int): The size of the batch to return from the iterator. Must be greater than 0.
+
+        Returns:
+            train (Iterator): An iterator for training values.
+            test (Iterator): An iterator for testing values.
+        """
+        if batch_size < 1:
+            raise ValueError("batch_size must be 1 or greater")
+
+        if random_state == 0:
+            raise ValueError("random_state must be greater than zero or None")
+
+        test_size, train_size = self.validate_test_size(test_size, train_size)
+
+        name = self._stream.name
+        variant = self._stream.version
+        stub = self._stream._stub
+        model = self._stream.model if hasattr(self._stream, "model") else None
+        if random_state is None:
```

Comment:
shouldn't this `random_state` conditional be a part of the first if check on line 533?

also it sets the `random_state` to zero, but that's an error state value in the first conditional

Reply:
also really good catch